.github/workflows/pr-slack-notification.yaml

```diff
@@ -0,0 +1,297 @@
+name: Fetch All Open and Recently Closed Pull Requests using cron job
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '30 15 * * 1-5'  # At 15:30 on every day-of-week from Monday through Friday(UTC Time), 9pm in IST and 11:30 AM in EST.
+
+jobs:
+  fetch_all_pull_requests_and_notify_using_condition_and_cron:
+    runs-on: ubuntu-latest
+
+    env:
+      REPO: ${{ github.repository }}
+      BRANCH: main
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      API_URL: https://api.github.com/repos/redhat-developer/lsp4ij/pulls
+      SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Fetch recent cache
+        run: |
+          gh extension install actions/gh-actions-cache
+
+          # Fetching list of cache keys........
+          cacheKeys=$(gh actions-cache list -R $REPO -B $BRANCH -L 100 | cut -f 1 )
```

Comment:
I guess this is limiting the list of cache keys returned to 100. Are we likely to hit this limit? What would happen if we do?

Reply:
@mrglavas , Allowed values are within this limit `1-100` , Only the latest 100 cache keys will be listed. Any older cache keys beyond this limit will not be included in` $cacheKeys`. I am hoping that last cron job key will not be beyond the limit of 100.

Reply:
@vaisakhkannan It sounds unlikely that we would hit the limit but it would be good to document that the workflow has this limitation.

Reply:
@mrglavas, I added the comment in the code